backend/app/modules/uq/service/preliminary.py

```diff
@@ -33,6 +37,22 @@ class PreliminaryUQ(Model):
     epoch0_whitelist: Epoch0Whitelist
     identity_call_whitelist: IdentityCallWhitelist
 
+    def retrieve(self, context: Context, user_address: str) -> Decimal:
+        """
+        Calculates UQ when the user hasn't allocated to any project yet.
+        If the user has allocated to any project, the UQ is based on the first allocation.
+        """
+        epoch_num = context.epoch_details.epoch_num
+
+        uq_score = get_uq_by_address(user_address, epoch_num)
+        if not uq_score:
```

Comment:
This seems to be a problem to me. Logic in allocation is similar:
```
        if not get_uq_by_user(user, context.epoch_details.epoch_num):
            score = self.uniqueness_quotients.calculate(context, user_address)
            save_uq(user, context.epoch_details.epoch_num, score)
```
so, user won't be able to change they uq score before allocation if they call this endpoint first. We should:

a) don't save it in this endpoint

b) add some indicator in the db if it's uq saved by allocation or not. And if it's saved by allocation then we can't change it anymore, otherwise it's still editable

Reply:
absolutely, changed